pkg/factory/risingwave_object_factory.go

```diff
@@ -1160,6 +1237,40 @@ func (f *RisingWaveObjectFactory) NewFrontendDeployment(group string, podTemplat
 	return mustSetControllerReference(f.risingwave, frontendDeployment, f.scheme)
 }
 
+func (f *RisingWaveObjectFactory) NewFrontEndCloneSet(group string, podTemplates map[string]risingwavev1alpha1.RisingWavePodTemplate) *kruiseappsv1alpha1.CloneSet {
+	componentGroup := buildComponentGroup(
+		f.risingwave.Spec.Global.Replicas.Frontend,
+		&f.risingwave.Spec.Global.RisingWaveComponentGroupTemplate,
+		group,
+		f.risingwave.Spec.Components.Frontend.Groups,
+	)
+	if componentGroup == nil {
+		return nil
+	}
+
+	restartAt := f.risingwave.Spec.Components.Frontend.RestartAt
+
+	podTemplate := f.buildPodTemplate(consts.ComponentFrontend, group, podTemplates, componentGroup.RisingWaveComponentGroupTemplate, restartAt)
+
+	f.setupFrontendContainer(&podTemplate.Spec.Containers[0], componentGroup.RisingWaveComponentGroupTemplate)
+
+	keepPodSpecConsistent(&podTemplate.Spec)
```

Comment:
nit: please also keep the comments aligned

```suggestion
	keepPodSpecConsistent(&podTemplate.Spec)
	
```